### ISSUE TITLE
[Suggestion] Ristretto Set may fail and it's the expected behavior

### ISSUE THREAD

**Issue by Yiling-J** [eko/gocache#206]
From Ristretto's Readme [FAQ](https://github.com/dgraph-io/ristretto#faq):
```
As for "shortcuts," the only thing Ristretto does that could be construed as one is dropping some Set calls
```

So both set and setTags may return false when using Ristretto. Maybe better to mention it in readme or do something in code.

And here is a simple parallel benchmark to demonstrate that:
```Go
func BenchmarkRistrettoSetParallel(b *testing.B) {
	ctx := context.Background()

	client, err := ristretto.NewCache(&ristretto.Config{
		NumCounters: 1000,
		MaxCost:     100,
		BufferItems: 64,
	})
	if err != nil {
		panic(err)
	}
	store := NewRistretto(client, nil)

	b.RunParallel(func(pb *testing.PB) {
		i := 0
		for pb.Next() {
			key := fmt.Sprintf("test-%d", i)
			value := []byte(fmt.Sprintf("value-%d", i))
			err := store.Set(ctx, key, value, lib_store.WithTags([]string{fmt.Sprintf("tag-%d", i)}))
			if err != nil {
				fmt.Print("x")
			}
			i++
		}
	})
}
```
you'll see some "x" when running it

**Comment by eko** [comment]
Agreed, and it is now documented in the README next to the store capability table: Ristretto is allowed to drop a `Set()` when its buffers are full — it is by design in its FAQ — in which case the store returns an error. `store.WithSynchronousSet()` is the way to make a write visible right after the call.

Two related Ristretto fixes shipped in `store/ristretto/v4.3.7`: setting a value with tags used to panic for any store type other than `[string, []byte]` (the tag list was written as `[]byte` whatever the value type), and the tag list started with an empty entry.

Don't hesitate to reopen this issue if the problem persists.
